src/js/components/form/FieldAutofocus.js

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import DOMUtils from "#SRC/js/utils/DOMUtils";
+
+let lockFieldAutofocus = false;
```

Comment:
The lockFieldAutofocus is needed to lock the autofocus functionality to the first element that triggers it.

In our page we may have multiple fields with the autofocus functionality. For example in my Services Form every time I click "ADD ENVIRONMENT VARIABLE" I would like to focus on the first input field entered. In this case, this is the Key input field. 

This can be easily done now if we wrap the input with the FieldAutofocus

```
<FieldAutofocus>
    <FieldInput
        name={`env.${key}.key`}
        type="text"
        value={env.key}
    />
</FieldAutofocus>
``` 

![image](https://user-images.githubusercontent.com/180432/27910937-9414f7b2-6257-11e7-8777-5d9b4909b45d.png)

We can wrap this new container component anywhere we would like to autofocus once the inner input/ textarea is mounted on the page.

The problem now is that if we have multiple components with the FieldAutofocus wrapper the document would focus on the last input that triggers a focus. 

In the example above if we click on the "Service" tab and go back to the "Environment" tab we would like to autofocus on the first Key input field and not the second one. To do this we set the lockFieldAutofocus to "true" and once we blur the component it is set to "false" again.